File: YTPlaylistSearcherWebApp/DTOs/PlaylistDTO.cs
namespace YTPlaylistSearcherWebApp.DTOs
{
    public class PlaylistDTO
    {
        public int ID { get; set; }
        public string PlaylistID { get; set; }
        public string PlaylistTitle { get; set; }
        public string ChannelOwner { get; set; }
        public IEnumerable<VideoDTO> Videos { get; set; }
    }
}

File: YTPlaylistSearcherWebApp/ClientApp/src/app/DTOs/PlaylistDTO.ts

export interface PlaylistDTO {
  id: number,
  playlistID: string;
  playlistTitle: string;
  channelOwner: string;
  videos: VideoDTO[];
}

export interface VideoDTO {
  id: number,
  videoID: string;
  title: string;
  channelTitle: string;
  description: string;
  thumbnail: string;
  publishedDate: string | null;
}
